spec/unit/network/http/request_spec.rb

```diff
@@ -63,44 +66,44 @@ def a_request(headers, body = "")
       it "raises HTTP 400 if the server doesn't specify a default" do
         request = a_request({})
         expect {
-          request.response_formatter_for([:json])
+          request.response_formatters_for([:json])
         }.to raise_error(bad_request_error, /Missing required Accept header/)
       end
 
       it "uses the server default" do
         request = a_request({})
-        expect(request.response_formatter_for([:json], 'application/json').name).to eq(:json)
+        expect(request.response_formatters_for([:json], 'application/json')).to eq([json_formatter])
       end
     end
 
-    it "selects the first format supported by the server" do
+    it "selects the first format supported by the server and accepted by the client" do
```

Comment:
I reworked this test to ensure multiple formats are returned in the accepted order, even when the supported order is different.